src/cryptonote_core/service_node_list.cpp

```diff
@@ -3286,10 +3255,41 @@ void service_node_list::state_t::update_from_block(
                         throw oxen::traced<std::runtime_error>{
                                 "Internal error: did not find state change tx data in blockchain database: {}"_format(
                                         fail)};
+
+                    // NOTE: Grab TX index of the L2 transaction it was originally mined in
+                    uint64_t tx_index = 0;
+                    try {
+                        cryptonote::block block =
+                                db.get_block_from_height(unconf.height_added, nullptr);
+                        bool found = false;
+                        for (size_t index = 0; index < block.tx_hashes.size(); index++) {
+                            if (block.tx_hashes[index] == txhash) {
+                                tx_index = index;
+                                found = true;
+                                break;
+                            }
+                        }
+                        if (!found)
+                            throw;
```

Comment:
naked throw outside of a catch block

Reply:
Doesn't it get caught by the one just a couple of lines below it? Or is my cpp-foo too weak and that's not how that works.

Reply:
Wasn't sure about this myself, but a cursory search found [this comment](https://stackoverflow.com/a/2288727) which quotes the standard:

> If no exception is presently being handled, executing a throw-expression with no operand calls `terminate()`.

If you were to `throw some_exception{"bla"};` here, it would indeed be caught just below.

Reply:
TIL, I have pushed a change to not do that.